mongo/gridfs/download_stream.go

```diff
@@ -226,6 +226,11 @@ func (ds *DownloadStream) GetFile() *File {
 func (ds *DownloadStream) fillBuffer(ctx context.Context) error {
 	if !ds.cursor.Next(ctx) {
 		ds.done = true
+		// Check for cursor error, otherwise there are no more chunks.
+		if ds.cursor.Err() != nil {
+			ds.cursor.Close(ctx)
```

Comment:
`Close` can return an error, but I'm not sure what to do with it here. Should it supersede the `ds.cursor.Err()`?

Reply:
I agree with your decision to ignore the error from `Close` here because we want to surface the error that was actually related to iteration. Let's make it more explicit by doing `_ = ds.cursor.Close()`, though.

Reply:
Done.